app/Scripts/boid.gd

```diff
@@ -13,26 +13,58 @@ export var follow_target = Vector2()
 var player : Node2D
 export var separation_threshold = 1.0
 
-
+var boids := [] 
 var velocity = Vector2()
 
 var enemy_target : bool = false
 var enemy_to_target : Node2D
-
 #each boid needs to keep an array of each other boid in the scene. 
 #might be optimized by outsourcing the logic for moving the boids to another
 #script that keeps track of all the boids and sends them their position each frame
-var boids = []
 
+######
+#Health Related Code
+onready var HpBar = $HPbar
+var health: float
+var MaxHealth := 100
+
+
+func DamageBoid(damage): # use this function when damaging the boid
+	health -= damage
+func HealthCalculations():
+	HpBar.value = health
+	if(health == MaxHealth):
+		HpBar.visible = false
+		
+	elif(health != MaxHealth):
+		
+		HpBar.visible = true
+	if health <= 0:
+		health = 0
+		emit_signal("BoidDied", self)
+
+		
+		PlayerStats.FollowingBoids.remove(PlayerStats.FollowingBoids.find(self))
+		PlayerStats.BoidsCollectedNum = len(PlayerStats.FollowingBoids)
+		self.queue_free()
+	if health >= MaxHealth:
+		health = MaxHealth
+######		
 func _ready():
+	HpBar.value = 100
+	health = MaxHealth
 	self.set_meta("Boid", false)
 	get_parent().get_node("Enemy").connect("_enemy_moused_over_true", self, "_enemy_moused_over_true")
 	get_parent().get_node("Enemy").connect("_enemy_moused_over_false", self, "_enemy_moused_over_false")
+	
 func _process(delta):
 	#find all boids in the normal process to keep accurate track of them, all
 	#other calculations will be done in physics process to keep them framerate independent
 	boids = find_all_boids()
-
+	HealthCalculations()
```

Comment:
we don' need to run this every frame, just at initialization and when damage is taken right?

Reply:
Yes but for some reason setget wasnt working

Reply:
do you want to figure out why it wasn't working?